### PR TITLE
Add ability to pass arguments via cli

### LINKED ISSUE #30: Consider moving config.yml as inputs to the CLI
**Is your feature request related to a problem? Please describe.**
It seems that keeping a config file separate from the process is currently not serving any purpose (?). Removing this loose dependency would allow users to easily call the process with the wanted settings from external applications without the need to materialize an yml file. Moreover, it may also allow users to launch parallel watchdog processes to watch multiple folders at the same time. This may be an important point if the same rig is being shared across different experiments that might log data to different places.

**Describe the solution you'd like**
Move the 3 variables in the config class:
https://github.com/AllenNeuralDynamics/aind-watchdog-service/blob/49092f270dbc675fc0e4fa028d6903ed6bf90407/src/aind_watchdog_service/models/watch_config.py#L8-L23

As variables to the executable CLI

**Describe alternatives you've considered**
Leave it as is. 

**Additional context**

### PR DESCRIPTION
**Pull request by bruno-f-cruz**
This PR allows passing arguments via the cli interface:

The current order of priorities is:
 - if a path is passed to `-c`, it will be used as the target to find the config yml file
 - if `-c` is not passed, and `-m` and `-f` are set, they will be used to construct the `WatchConfig` model. If `-m` or `-f` are set, both must be set
 - Finally, if no arguments are passed (fully backward compatible) the application will default to try to find the file via the environment variables.

The PR also adds an initial check (fail early) if the target yml file doesn't exist.

``` cmd
python .\src\aind_watchdog_service\main.py --help
usage: main.py [-h] [-c CONFIG_PATH] [-f FLAG_DIR] [-m MANIFEST_COMPLETE] [-w WEBHOOK_URL]

Watchdog service

options:
  -h, --help            show this help message and exit
  -c CONFIG_PATH, --config-path CONFIG_PATH
                        Configuration file for watchdog service. Takes precedence over environment variable WATCH_CONFIG and other arguments
  -f FLAG_DIR, --flag-dir FLAG_DIR
                        Directory for watchdog to poll
  -m MANIFEST_COMPLETE, --manifest-complete MANIFEST_COMPLETE
                        Manifest directory for triggered data
  -w WEBHOOK_URL, --webhook-url WEBHOOK_URL
                        Teams webhook url for user notification

```

Fixes #30

### CODE REVIEW

**Comment by bruno-f-cruz** [comment]
@arielleleon @jtyoung84 
I have fixed a few edge cases on this PR. 

During my tests I came across #38.

Finally, is the `-noconsole` flag for the build necessary? The CLI works find without the `stdout`, but I find it would be great to be able to report the status of ongoing processes via the command line. I guess the fear is that people may close the process inadvertently by closing the console window? 

Anyway the PR is merged and as far as I can tell is fully backward compatible :)

**Review by arielleleon — APPROVED** [review]
Tested this out locally and works great. I can add some tests for the cli once the review is finished

File: src/aind_watchdog_service/main.py [hunk 1]
```diff
@@ -82,24 +84,88 @@ def start_service(self) -> None:
...
+        "-w", "--webhook-url", type=str, help="Teams webhook url for user notification"
+    )
+
+    args = parser.parse_args()
+
+    if args.config_path:
+        args.flag_dir = None
+        args.manifest_complete = None
+
+    if (args.flag_dir is None) ^ (args.manifest_complete is None):
```
[the remark this answers is not in the document]
Comment (bruno-f-cruz) [reply src/aind_watchdog_service/main.py:124]: Assuming IT policies are compatible, I would prob use %appdata% under a folder called `aind-watchdog-service`. I think we could still ship the CLI with it for now and open an issue with this idea for a next release
